crypto/test/tests.go

```diff
@@ -63,17 +39,52 @@ func SignedDataComponent(tb testing.TB, signer neofscrypto.Signer, cmp SignedCom
 	require.True(tb, cmp.VerifySignature())
 }
 
-// SignedDataComponentUser tests [SignedComponentUserSigner] for valid data generation by SignedData function.
-func SignedDataComponentUser(tb testing.TB, signer user.Signer, cmp SignedComponentUserSigner) {
-	data := cmp.SignedData()
+type failedSigner struct {
+	neofscrypto.Signer
+}
 
-	sig, err := signer.Sign(data)
-	require.NoError(tb, err)
+func (x failedSigner) Sign([]byte) ([]byte, error) { return nil, errors.New("[test] failed to sign") }
 
-	static := neofscrypto.NewStaticSigner(signer.Scheme(), sig, signer.Public())
+// FailSigner wraps s to always return error from Sign method.
+func FailSigner(s neofscrypto.Signer) neofscrypto.Signer {
+	return failedSigner{s}
+}
 
-	err = cmp.Sign(user.NewSigner(static, signer.UserID()))
-	require.NoError(tb, err)
+// Signature returns random neofscrypto.Signature.
+func Signature() neofscrypto.Signature {
+	sig := make([]byte, 64)
```

Comment:
It was a question which I forget to ask many time. Do we want to move such `magic` numbers inside packages they are belongs to? I mean this `64` to _neofscrypto.SignatureBytesLength_ for example.

[Another example](https://github.com/nspcc-dev/neofs-sdk-go/blob/75bcf1dc149b5594be1a117be01b49e5c2244d48/crypto/ecdsa/public.go#L22) of magic numbers, this `33` would be perfect to has a public const, because in the code in projects we have a lot of `pubKey = make([]byte, 33)`
`pubKey = make([]byte, ecdsa.PublicKeyLength)` looks much better for me.

Yes, in some cases it could be inconvenient to export one more package, but in general case it would ok

Reply:
code should not do `pubKey = make([]byte, 33)`: https://pkg.go.dev/github.com/nspcc-dev/neofs-sdk-go@v1.0.0-rc.12/crypto#PublicKey has method to get len

 actually i dont wanna make it const, forgot to add some randomness, thx anyway